### ISSUE TITLE
Read random seed from snapshot if random seed is not provided

### ISSUE THREAD

**Issue by ctlaltdefeat** [catboost/catboost#308]
Problem: When snapshotting, I am never able to resume from the file.
I always get: ```Can't load progress from file: catboost_progress exception: (TCatboostException) catboost/libs/algo/learn_context.cpp:255: Saved model's Params are different from current model's params``` even when just stopping training and calling fit again using the same instance of the model.

Also, related questions:
1. The snapshot file doesn't even get created when using GPU. Is there a possibility to fix this?
2. Is there some other way, perhaps not with snapshots, to just fit some number of iterations, stop, and then fit some more?

catboost version: 0.7.2
Operating System: Linux

**Comment by Noxoomo** [comment]
1) Could you please provide options you used for training? And do you use python wrapper?
2) GPU saves snapshot every 10 minutes. If training if less, then 10 minutes, there'll be no snapshot (but I didn't check it for python package, may be there some bug)

**Comment by ctlaltdefeat** [comment]
Yes, using Python. The example in the docs also raises this issue:

```
import numpy 
from catboost import CatBoostRegressor
dataset = numpy.array([[1,4,5,6],[4,5,6,7],[30,40,50,60],[20,15,85,60]])
train_labels = [1.2,3.4,9.5,24.5]
model = CatBoostRegressor(learning_rate=1, depth=6, loss_function='RMSE', save_snapshot=True, snapshot_file='asdf', iterations=500)
fit_model = model.fit(dataset, train_labels)
print(fit_model.get_params())
```

And when running it again: ```Can't load progress from file: catboost_progress exception: (TCatboostException) catboost/libs/algo/learn_context.cpp:255: Saved model's Params are different from current model's params```

2. Is this because it's expensive to load from the GPU's memory? Is there a possibility to modify this time? Ten minutes is way too much for me. 

I also noticed that there's a delay when calling fit until the above error shows up. I would have thought that resuming would be instantaneous. My use case is that I would like to search over many different sets of hyperparameters, so I would like to train all the models for some iterations, pause, and then continue with only a few of the models. If resuming is not almost immediate then I may need a different approach.

**Comment by ctlaltdefeat** [comment]
@Noxoomo But if the models share a common Pool and just have different hyperparameters, does memory still need to be freed and reallocated? That's the usecase I have.

**Comment by Noxoomo** [comment]
@ctlaltdefeat 
It's not true in general.  We don't use Pool directly during learning, we convert it to more efficient  structure.
Some hyperparameters will change internal data representation (feature_border_count, CTR parameters, one_hot_max_size, ignored_features will require rebuilding interior data)

**Comment by annaveronika** [comment]
This is a bug, it will be fixed in next release. For now you can provide random_seed when start training, and it should work fine.

**Comment by annaveronika** [comment]
Fix is on github now, please build from source using this instruction:
https://tech.yandex.com/catboost/doc/dg/installation/python-installation-method-build-from-source-linux-macos-docpage/

It will be on pypi in the next release.
